lib/active_storage_validations/aspect_ratio_validator.rb

```diff
@@ -24,11 +26,12 @@ def validate_each(record, attribute, _value)
         changes = record.attachment_changes[attribute.to_s]
         return true if changes.blank?
 
+        options = unfold_procs(record, self.options, AVAILABLE_CHECKS)
         files = Array.wrap(changes.is_a?(ActiveStorage::Attached::Changes::CreateMany) ? changes.attachables : changes.attachable)
 
         files.each do |file|
           metadata = Metadata.new(file).metadata
-          next if is_valid?(record, attribute, metadata)
+          next if is_valid?(record, attribute, metadata, options)

```

Comment:
This `options` is an instance variable (or at least it should be). So passing this should be unnecessary. If the unfolding of the options has to be done, do that in the location where you need it (in the private method).

Reply:
No it is not the instance variable, it's the local variable `option` which was masked above. Which I will rename to `flat_options` to make it clearer.